R/generate_eval_data.R

```diff
@@ -0,0 +1,98 @@
+#' Generate evaluation data for a hub
+#'
+#' @param hub_path A path to the hub.
+#' @param config_path A path to a yaml file that specifies the configuration
+#' options for the evaluation.
+#' @param out_path The directory to write the evaluation data to.
+#' @param oracle_output A data frame of oracle output to use for the evaluation.
+#'
+#' @export
+generate_eval_data <- function(hub_path,
+                               config_path,
+                               out_path,
+                               oracle_output) {
+  config <- read_config(hub_path, config_path)
+  for (target in config$targets) {
+    generate_target_eval_data(hub_path, config, out_path, oracle_output, target)
+  }
+}
+
+
+#' Generate evaluation data for a target
+#'
+#' @inheritParams generate_eval_data
+#' @param config The configuration options for the evaluation.
+#' @param target The target to generate evaluation data for. This is one object
+#' from the list of targets in the config, with properties "target_id",
+#' "metrics", and "disaggregate_by".
+#'
+#' @noRd
+generate_target_eval_data <- function(hub_path,
+                                      config,
+                                      out_path,
+                                      oracle_output,
+                                      target) {
+  target_id <- target$target_id
+  metrics <- target$metrics
+  disaggregate_by <- target$disaggregate_by
+  eval_windows <- config$eval_windows
+
+  task_groups_w_target <- get_task_groups_w_target(hub_path, target_id)
+  metric_name_to_output_type <- get_metric_name_to_output_type(task_groups_w_target, metrics)
+
+  for (eval_window in eval_windows) {
+    model_out_tbl <- load_model_out_in_window(hub_path, target$target_id, eval_window)
+
+    # calculate overall scores followed by scores disaggregated by a task ID variable.
+    for (by in c(list(NULL), disaggregate_by)) {
```

Comment:
Minor: this makes the assignment a bit more explicit so that the for loop is easier to read. 

```suggestion
  # adding `NULL` at the beginning will calculate overall scores
  disaggregate_by <- c(list(NULL), as.list(target$disaggregate_by))
  eval_windows <- config$eval_windows

  task_groups_w_target <- get_task_groups_w_target(hub_path, target_id)
  metric_name_to_output_type <- get_metric_name_to_output_type(task_groups_w_target, metrics)

  for (eval_window in eval_windows) {
    model_out_tbl <- load_model_out_in_window(hub_path, target$target_id, eval_window)

    # calculate overall scores followed by scores disaggregated by a task ID variable.
    for (by in disaggregate_by) {
```